recipe/meta.yaml

```
@@ -1,119 +1,121 @@
-    - libpng
+    {% if cuda_major >= 12 %}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - libcublas-dev                          # [build_platform != target_platform]
+    - libcusolver-dev                        # [build_platform != target_platform]
+    - libcusparse-dev                        # [build_platform != target_platform]
+    {% endif %}
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+    - pytorch ={{ compatible_pytorch }}.*={{ torch_proc_type }}*  # [build_platform != target_platform]
   host:
     - python
     - pip
+    - setuptools
     - wheel
-    - numpy {{ numpy }}
-    - pytorch 2.0.1
-    # libpng and libjpeg must be available at compilation time
-    - jpeg {{ jpeg }}  # [not win]
+    - cudnn                                      # [cuda_compiler_version != "None"]
+    {% if cuda_major >= 12 %}                    # [cuda_compiler_version != "None"]
+    - libcublas-dev                              # [cuda_compiler_version != "None"]
+    - libcusolver-dev                            # [cuda_compiler_version != "None"]
+    - libcusparse-dev                            # [cuda_compiler_version != "None"]
+    {% endif %}                                  # [cuda_compiler_version != "None"]
+    - cudatoolkit {{ cuda_compiler_version }}    # [cuda_compiler_version == "11.8"]
+    - cuda-version {{ cuda_compiler_version }}   # [cuda_compiler_version != "None"]
+    # split off image/video into separate outputs?
+    - jpeg {{ jpeg }}
     - libpng {{ libpng }}
-    # GPU requirements
-    - cudatoolkit {{ cudatoolkit }}  # [pytorch_variant == "gpu"]
-    # 2022/08/25: Not requiring ffmpeg because torchvision has bugs with certain ffmpeg versions, 
-    # see https://github.com/pytorch/vision/issues/5419#issuecomment-1193483864
-    # and https://github.com/conda-forge/torchvision-feedstock/commit/4ec4b5f4eb4889dbb1f8da34662ea622bb4b3828
-    # - ffmpeg
+    - ffmpeg
+    - numpy
+    # exclude 8.3.0 and 8.3.1 specifically due to pytorch/vision#4146, python-pillow/Pillow#5571
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
+    #- libtorch
+    - pytorch ={{ compatible_pytorch }}.*=*{{ torch_proc_type }}*
+    - requests
   run:
-    # GPU requirements
-    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [pytorch_variant == "gpu"]
-    - {{ pin_compatible('numpy') }}
     - python
-    - pillow >=5.3.0,!=8.3.*
-    - pytorch
+    - pytorch ={{ compatible_pytorch }}.*=*{{ torch_proc_type }}*
+    - {{ pin_compatible('cudnn') }}           # [cuda_compiler_version != "None"]
+    - cudatoolkit {{ cuda_compiler_version }}    # [cuda_compiler_version == "11.8"]
+    - cuda-version {{ cuda_compiler_version }}   # [cuda_compiler_version != "None"]
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
```

```suggestion
    - pillow >=5.3.0,!=8.3.0,!=8.3.1
    - numpy <2.0a0
```
If you end up removing the numpy2 compat patch. Otherwise just numpy.

---

in progress